### ISSUE TITLE
Analog stick can be more precise, like M64 system.

### ISSUE THREAD

**Issue by Dougzen** [hrydgard/ppsspp#14410]
Hello again.

If you test the M64 analog sensitivity you will noticed one difference:

When you push the area around the stick, he goes automatically to the direction.

In the PPSSPP case, you need drag the analog since your center to go to some direction.

I think that the M64 way is better in this case, will be nice find a way to put this in PPSSPP too.

Thank you :D

**Comment by Dougzen** [comment]
Update:

I found out more things:

You guys already have this function.

But the range area is too short and the neutral point of the analog (center of analog) is too big.

This makes the range of the analog be too short instead large.

**Comment by LunaMoo** [comment]
Not sure if I understand you, but analog deadzone is configurable in PPSSPP.

 Games itself however tend to have large deadzone to analog stick on their own due to PSP thumbstick being crappy. Most games which do use it, actually use it in digital mode with added acceleration, there's nothing emulator can do about it, through users themselves can patch games to support analog properly, a nice example would be my right analog patch for MGS PW.

**Comment by Dougzen** [comment]
This is the point, isn't configurable enough.

Just testing the M64Plus emulator for Android to see the difference, using the M64 analog you can push a large deadzone without DRAG the analog since your center of stick.

The factor here is the "drag" difference, while in M64 the "circle of distance" in the analog is big, in the PPSSPP case he is too small, making you practically have to drag the analog stick since of your center position.

Is a little thing, but have a lot difference in gameplay experience.

**Comment by LunaMoo** [comment]
If you mean touch controls, you can simply enlarge the analog inside the "edit touch control layout..." screen for it to be bigger which in result does make it more sensitive.

 But it will still hold that majority of PSP games didn't cared to use analog in a nice fashion due to PSP being a portable system with a crappy thumbstick and not a full size gamepad with stick that you could accurately control and this can't be changed on the emulator side.

**Comment by Dougzen** [comment]
Making the analog stick bigger is useless and I will explain why:

What I trying to say is this:
Imagine 3 things:

1 - If you could play Mario 64 using the PPSSPP analog stick, will be worst than play using the M64 analog stick.

2 - This happens because the structure of the analog stick is splited in 2 forms: You have the stick and the head of stick:

2.1 - The head of stick is always big.
2.2 - The stick isn't always big.

2.3 - The M64 analog stick have a slim normal stick and have a "normal head". This gives more precision to using the analog stick. The area is bigger.

2.4 The PPSSPP analog stick have a big normal stick and a big "normal head". This gives less precision to using the analog stick. The area is shorter.

You guys can understand my point now?

3 - Now imagine the 3 different speeds of Mario walking in Mario 64: Slow, Medium and Run.

Using the M64 way you can test all the speeds easily.

But using the PPSSPP way you can test all too. But is harder doing this, because the area is too short.

To solve this problem is very simple.

The developers just need make the "normal stick" part smaller like the M64Plus analog stick. 

Or just copy the code of the M64 analog stick and put the PPSSPP skin. The M64 code is open like PPSSPP so there's no problem in copy.

**Comment by Dougzen** [comment]
> But it will still hold that majority of PSP games didn't cared to use analog in a nice fashion due to PSP being a portable system with a crappy thumbstick and not a full size gamepad with stick that you could accurately control and this can't be changed on the emulator side.

But 2 big games actually uses the analog stick:

Kingdom Hearts Birth by Sleep
Gran Thief Auto: Vice City.

**Comment by Dougzen** [comment]
You guys can see my point now?

It's really important do this changes, because this will bring more users and players to the PPSSPP community.

**Comment by Dougzen** [comment]
@hrydgard Do you can give me a answer about this?
You think that is a important thing to do or not?

I have anxiety problems and this discussion is just ending with my peace.

**Comment by LunaMoo** [comment]
It's not useless to make analog bigger in touch controls and it does make it easier to make fine adjustments, I used it myself for aiming in one FPS games I patched for proper analog control.

 What you need to understand is that PSP never really had analog stick, it was a delicate, non accurate thumbstick sliding around and breaking easily, game devs mostly knew that so most PSP games does not use it in a way you want, it's usually:
 - not using thumbstick at all,
 - using it, but in digital mode only,
 - using it in analog mode, but includes functions in the game code that dumb it down to ignore most of the analog range etc.

 I don't know if other people who patched in analog support made a proper analog range by themselves or just re-mapped reusing existing game code, but I did a few of such patches myself and the difference between full analog and what's in most PSP games I know as a fact is huge, doesn't matter if using real gamepad or touch controls.

**Comment by Dougzen** [comment]
Why I will not use the analog stick if he already exist?

Playing with the a real PSP analog is different that playing the analog in touchscreen. Because every real and physical controller it's better than any touch controller.

I don't understand why people must be forbidden of use a new type of analog, If can be just a optional version of him.

Isn't affect nothing of what you will play, if you can change between the actual and the new analog that I am proposing.

Man, it's hard talk to you, I just want improvent this function to all players, and you is against me with all your forces. It's a optional version and not a silly issue.

**Comment by Dougzen** [comment]
That isn't make any sense.

I trying to issue a new and optional function and you just denied all the arguments, because you want.

I want the main developer answer and not people shooting me, going against me.

If you want help me, just says something useful and not demote or demean my issue.

**Comment by Dougzen** [comment]
I hope that @hrydgard see this thread and I will close this.

Because you make a big absurd here trolling my issue just because you want.

**Comment by LunaMoo** [comment]
I'm not against you in any way, get over it. I was also in no way negative, I even opened myself to you about also suffering from social anxiety disorder and told you to stop using it as an excuse.

 All I'm saying PSP games does not have analog support comparable to full size console games and when you said making touch analog bigger doesn't help I said you're wrong as it does make it much easier to get fine adjustments in game patched to actually use it saying most PSP games does NOT work that way. I know that probably better than most people because I actually spent a ton of time patching a few example games for analog controls hoping community will be interested and people will start making similar patches improving everyone's experience, it just never happened.

**Comment by Dougzen** [comment]
It's funny you say that the most of games NOT work that way.
Because the PPSSPP already had a single configuration to play God of War.

Because God of War is a amazing game and people wanted play better.

You practically is saying that:

Kingdom Hearts Birth by Sleep
And
Gran Thief Auto Vice City
Are horrible games to make better the analog stick function.

I am saying, why not make this improvement when we have a amazing M64 example of analog stick?
All that the developer need to do is copy the code and put the PPSSPP skin.
And after all put how is was a option to change the analog.

It's simple.

**Comment by LunaMoo** [comment]
I don't understand what you're saying about GOW and KH & GTA, but if you're saying the first works fine while the latter sucks, then yeah it would just confirm what I told you that most PSP games doesn't have a real / full range analog stick support.

 Also I might even not understand what you're requesting, because I see it like that:
"Emulator of full size console has a better working analog stick, copy it from it, because PPSSPP sucks."

 And basically tell you full size consoles had games making advantage of full size gamepads while PSP games did not and I know as a fact that most of them are pretty iffy about analog range if they even use a range.

**Comment by Dougzen** [comment]
So let's close this entire PPSSPP site, because nobody can make new issues.
We already have a perfect emulation and we don't need new issues and changes.
It's just a handheld console, let's stop of making improvements.

This is your logical.

I am not saying that the current is sucks, I am just saying that can be better.

**Comment by LunaMoo** [comment]
No, that's not my logic, calm your tits and read what I wrote above, but since you're now trolling, I'll just ignore this issue.

 I tried to explain why analog control between PSP(a handheld with no analog stick) vs a different console(with full sized gamepads) was a bad idea.

 You claim I attack you which is apparently something anyone nowadays can take up from their ass as long as then they claim they have anxiety in the end to call up knights, but you know what? I had suffered from anxiety disorder for a very long time and I could not even admit I had it because I was affraid of how people will see me afterwards, so now I think you don't really suffer from anything, but using it just to make excuses when you missunderstand a discussion going into direction you don't like and that's just disgusting and just bad for anyone who trully suffers from it.

**Comment by Dougzen** [comment]
Everyone who goes against you commit a excuse.

Do you have a idea of how absurd is you not being the creator of the PPSSPP and delegitimizing the other users issues?
People comes here to make issues and wait the @hrydgard answer or not.

If you agrees with my issue, you send a message and say something good.
If you think that my idea sucks, just not respond and save your lifetimes.

You live in society like this.
Not saying that all that you don't like it's sucks directly to the people.

Respect the others, instead saying that is just a excuse, or a sucks issue.
It's the minimal that you can do to live in society.

**Comment by hrydgard** [comment]
PPSSPP's analog stick calibration indeed needs work and better options. It's fine to keep this open.

**Comment by Dougzen** [comment]
> PPSSPP's analog stick calibration indeed needs work and better options. It's fine to keep this open.

Thank you so much Henry :D
This video shows exactly what I was saying.

In this case is a EPSXE emulator, the code is locked but the example it's nice.

https://youtu.be/WmRPkiPRY-A

**Comment by Calinou** [comment]
> PPSSPP's analog stick calibration indeed needs work and better options. It's fine to keep this open.

Out of curiosity, does https://github.com/hrydgard/ppsspp/pull/14596 resolve this? Not that I personally need it, but I was wondering this while reading the latest release notes.
